### PR TITLE
Add llama-index-readers-file

### PR DESCRIPTION
**Pull request by pavelzw**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

depends on https://github.com/run-llama/llama_index/pull/12406
CC @jan-janssen

### CODE REVIEW

**Comment by moritzwilksch** [comment]
#25905

**Comment by pavelzw** [comment]
depends on #25905

**Comment by jan-janssen** [comment]
I just merged https://github.com/conda-forge/pymupdf-feedstock/pull/5 - so once the package its available it should be possible to build this one.

**Comment by jan-janssen** [comment]
@conda-forge-admin, please restart ci

**Comment by jan-janssen** [comment]
Windows and MacOs are still missing `pymupdf`, still this package can remain `noarch`: 
```
Encountered problems while solving:
  - nothing provides requested pymupdf <2.0.0,>=1.23.21
```

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
@conda-forge/help-python

**Comment by jan-janssen** [comment]
@pavelzw I think this pull request is ready to be merged - can you switch it from draft to review ?

File: recipes/llama-index-readers-file/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - python >=3.8,<4.0
+    - poetry-core
+    - pip
+  run:
+    - python >=3.8.1,<4.0
+    - llama-index-core >=0.10.1,<0.11.0
+    - beautifulsoup4 >=4.12.3,<5.0.0
+    - pypdf >=4.0.1,<5.0.0
+    - striprtf >=0.0.26,<0.0.27
+    - pymupdf <2.0.0,>=1.23.21
```
Comment (pavelzw) [inline recipes/llama-index-readers-file/meta.yaml:31]: Theoretically this should only be a run export but the poetry metadata doesn't work correctly...
https://github.com/run-llama/llama_index/issues/12697

```suggestion
    # remove once https://github.com/run-llama/llama_index/issues/12697 is resolved
    - pymupdf <2.0.0,>=1.23.21
```

Comment (pavelzw) [reply recipes/llama-index-readers-file/meta.yaml:31]: i'll add this comment in the generated feedstock afterwards @xhochy 😄

Comment (pavelzw) [reply recipes/llama-index-readers-file/meta.yaml:31]: https://github.com/conda-forge/llama-index-readers-file-feedstock/commit/d4013ecaccc863eaf8a1ae7468c45290fc38ee35

**Pull request merged**